docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,327 @@
+workflows driven by parametrized notebooks executed with  `papermill`_. (see *Outlook* 
+section below) Moreover, some Python scripts have been added recently to perform `benchmarks`_
+of Gammapy, surely we could rewrite some of all of these benchmarks to use the high-level interface. 
+
+We also have some *high-level analysis* classes in the API that concatenate several atomic 
+actions and provide rough estimated results for more complex processes. (i.e. 
+`SpectrumAnalysisIACT`_, `LightCurveEstimator`_) These classes would serve as a basis to 
+design and prototype some of the tools of the high-level interface.
+
+
+Proposal
+========
+
+We will develop a high-level interface Python API which uses a params-values configuration file 
+defined by the user. This API would be used in Python scripts, notebooks or in IPython sessions
+to perform simple and most common IACT analysis.
+
+We see then two main options on how to use the High Level Interface API:
+
+- Within an IPython session or notebook, mostly dealing with a manager object to perform specific tasks in an **interactive** analysis process
+- In a Python script or notebook, declaring the orchestration of the tasks with a manager object for an **automatized** process
+
+This high-level interface API is similar to what it is done in `Fermipy`_ or HAP in HESS, including
+also the options to save and recover session states, as well as serialization of intermediate data
+products and logging. It is flexible enough to allow the user to work with the API at any stage of the
+analysis process, and not only from the start to the very end or in automatized process.
+
+**Use cases**
+
+The use cases covered are in the scope of a single analysis and model, not parametrized variations in a 
+multidimensional grid space of variables, and within a single region (e.g. 10 deg region with 5 sources).
+
+The main use cases for analysis to be covered are:
+
+- 3D map analysis
+- 2D map analysis
+- 1D spectrum analysis
+
+Including the main methods for data reduction, modelling and fitting:
+
+- On vs on/off data reduction
+- Different background models
+- Joint vs stacked likelihood fitting
+- Diagnostics (residuals, significance, TS)
+- Spectral points
+
+Making a SED may be the final part of the analysis, as many SED methods require the full model and 
+all energy data.
+
+**Configuration file**
+
+The configuration file will be in YAML format exposing the parameters and values needed for each one 
+of the tasks involved in the analysis process. To generate the config file, we could add a 
+``gammapy analysis config`` command line tool which dumps the config file with all lines commented
+out, and the users can then uncomment and fill in the parameters and values they care about. As an 
+alternative users could copy & paste from a config file example in the docs. We will develop a 
+schema and validate / give good error messages on read.
+
+Prototype configuration file.
```

Comment:
@Bultako - Suggest to make it clearer with a few words that this is an example config file just to illustrate some things it might contain and how YAML looks. What you want to avoid is that PIG discussion focuses in on the schema of the config file, so by saying that this is just an example and the schema will be developed iteratively over the coming months, also taking user feedback after v0.14 into account, you make the PIG less controversial and I think people will more comment on the basic idea of the proposal.

Reply:
Ok, done.